warehouse/accounts/models.py

```diff
@@ -191,6 +191,28 @@ def can_reset_password(self):
             ]
         )
 
+    @property
+    def can_register_pending_oidc_providers(self):
+        if self.is_superuser or self.is_moderator or self.is_psf_staff:
+            # Special users (superusers, moderators, PSF staff) can always
+            # register pending OIDC providers.
+            return True
+
+        # Ordinary users can register pending OIDC providers under the following
+        # conditions:
+        #  * Must have a verified primary email
+        #  * Must have 2FA enabled
+        #  * Must be at least 90 days old
+        # We don't place any restrictions on how many pending OIDC providers
+        # an acceptable user may register, since actual project creation
+        # is done on a "TOFU-like" basis, just like normal project creation.
+        ninety_days_ago = datetime.datetime.now() - datetime.timedelta(days=90)
+        return (
+            self.has_primary_verified_email
+            and self.has_two_factor
+            and self.date_joined <= ninety_days_ago
+        )
```

Comment:
I'm thinking this might be too restrictive. I think primary verified email is probably acceptable (you can't do much without one anyways) but the other two are going to prevent a lot of legitimate folks from using it and will probably be frustrating for them.

I think we can just remove those and add/modify this if we experience significant abuse.

Reply:
Sounds good to me! I'll drop these restrictions; that should also make the ability to report the failure reason a little simpler.

Reply:
c11015a305182818c48bd6079ae570e4e37799c8